### PR TITLE
chore: Support a TRPC client in App Router Server Components

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- As mentioned in this [comment](https://github.com/trpc/trpc/discussions/3185#discussioncomment-4167473) from a maintainer of `trpc` package, Next 13 is currently not compatible with trpc.
- This PR is an attempt to introduce a workaround to make it compatible with Next 13 via methods mentioned in [article 1](https://codevoweb.com/setup-trpc-server-and-client-in-nextjs-13-app-directory/) and [article 2](https://levelup.gitconnected.com/next-js-13-trpc-a-match-made-in-heaven-9764fadd1084).

## Requirement/Documentation

- [comment](https://github.com/trpc/trpc/discussions/3185#discussioncomment-4167473) from a maintainer of `trpc` package
- [article 1](https://codevoweb.com/setup-trpc-server-and-client-in-nextjs-13-app-directory/)
- [article 2](https://levelup.gitconnected.com/next-js-13-trpc-a-match-made-in-heaven-9764fadd1084).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: apps/web/next.config.js [hunk 1]
```diff
@@ -211,6 +211,8 @@ const nextConfig = {
       ...config.resolve.fallback, // if you miss it, all the other options in fallback, specified
       // by next.js will be dropped. Doesn't make much sense, but how it is
       fs: false,
+      // @TEMP
+      "superagent-proxy": false,
```
Comment (grzpab) [inline apps/web/next.config.js:215]: what's that for?

File: packages/lib/server/i18n.ts [hunk 2]
```diff
@@ -1,18 +1,5 @@
...
   const create = async () => {
-    const { _nextI18Next } = await serverSideTranslations(locale, [ns]);
-    const _i18n = i18next.createInstance();
-    _i18n.init({
-      lng: locale,
-      resources: _nextI18Next?.initialI18nStore,
-      fallbackLng: _nextI18Next?.userConfig?.i18n.defaultLocale,
-    });
-    return _i18n;
+    return () => "";
```
Comment (grzpab) [inline packages/lib/server/i18n.ts:3]: I take this function could be removed?

File: apps/web/utils/trpc-provider.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+          },
+        }),
+      ],
+      transformer: superjson,
+    })
+  );
+
+  return (
+    <trpc.Provider client={trpcClient} queryClient={queryClient}>
+      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
```
Comment (DmytroHryshyn) [inline apps/web/utils/trpc-provider.tsx:122]: I think we need to add HydrationClient here so we can pass trpcState from server to it

File: apps/web/next.config.js (hunk 1 above)
Comment (hbjORbj) [reply apps/web/next.config.js:215]: Discussed in slack. Without it, code fails to compile.

File: apps/web/utils/trpc-provider.tsx (hunk 3 above)
Comment (hbjORbj) [reply apps/web/utils/trpc-provider.tsx]: addressed

**Review by DmytroHryshyn — APPROVED** [review]

File: apps/web/app/event-types-1/layout.tsx [hunk 4]
```diff
@@ -5,14 +5,18 @@ import { getLayout } from "@calcom/features/MainLayout";
...
   children: ReactElement;
 };
 
-export default function EventTypesLayout({ children }: EventTypesLayoutProps) {
+export default async function EventTypesLayout({ children }: EventTypesLayoutProps) {
   const h = headers();
   const nonce = h.get("x-nonce") ?? undefined;
 
+  const x = await serverClient.testing.greeting();
+  console.log(x);
```
Comment (hbjORbj) [inline apps/web/app/event-types-1/layout.tsx:19]: this is successfully logged

File: packages/trpc/server/routers/_app.ts [hunk 5]
```diff
@@ -12,6 +14,11 @@ import { viewerRouter } from "./viewer/_router";
  */
 export const appRouter = router({
   viewer: viewerRouter,
+  testing: router({
```
Comment (hbjORbj) [inline packages/trpc/server/routers/_app.ts:17]: will remove after review is done

File: apps/web/app/event-types-1/layout.tsx (hunk 4 above)
Comment (hbjORbj) [reply apps/web/app/event-types-1/layout.tsx:19]: will remove after review is done

File: packages/lib/server/i18n.ts (hunk 2 above)
Comment (hbjORbj) [reply packages/lib/server/i18n.ts]: addressed

**Pull request merged**